Core/Graphics/Source/DeviceImpl.cpp

```diff
@@ -70,6 +62,29 @@ namespace Babylon::Graphics
         UpdateBgfxResolution();
     }
 
+    void DeviceImpl::SetMSAA(uint8_t MSAASamples)
+    {
+        std::scoped_lock lock{m_state.Mutex};
+        m_state.Bgfx.Dirty = true;
+        auto& init = m_state.Bgfx.InitState;
+        init.resolution.reset &= ~BGFX_RESET_MSAA_MASK;
+        switch (MSAASamples)
+        {
+            case 2:
+                init.resolution.reset |= BGFX_RESET_MSAA_X2;
+                break;
+            case 4:
+                init.resolution.reset |= BGFX_RESET_MSAA_X4;
+                break;
+            case 8:
+                init.resolution.reset |= BGFX_RESET_MSAA_X8;
+                break;
+            case 16:
+                init.resolution.reset |= BGFX_RESET_MSAA_X16;
+                break;
+        }
```

Comment:
Maybe we should throw here if someone passes in an invalid value? Or use an enum for this like I mentioned for the contract (in which case, there should be an assert here)?

Reply:
I had (and still have) a lot of hesitation using an enum. What bugs me with an enum is the number of definitions and switch all along the way:
With BRN Android, there would be an enum in JS, in Java and C++ and a switch/case conversion each time. Whereas a number of samples is just an integer that can be forwarded easily.